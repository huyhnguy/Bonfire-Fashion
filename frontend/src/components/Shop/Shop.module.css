.shop {
    display: flex;
    flex-wrap: wrap;
    gap: 5rem 1rem;
    justify-content:space-evenly;
    margin: 0 2rem 2rem 2rem;
}
