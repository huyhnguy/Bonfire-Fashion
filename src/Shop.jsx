import Navbar from "./Navbar"
import Product from "./Product"
import { useEffect, useState } from "react"

export default function Shop() {
    const [products, setProducts] = useState(null);

    useEffect(() => {
        fetch('https://fakestoreapi.com/products')
            .then(res=>res.json())
            .then(json=> setProducts(json))
    }, []);

    return (
        <>
            <Navbar />
            <h1>Hi! This is the shop page.</h1>
            { products &&
                products.map(product => <Product imgSrc={product.image} key={product.id} />)
            }
        </>
    )
}