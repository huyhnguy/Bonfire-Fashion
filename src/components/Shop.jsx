import Navbar from "./Navbar"
import Product from "./Product/Product"
import { useEffect, useState } from "react"

export default function Shop() {
    const [products, setProducts] = useState(null);

    useEffect(() => {
        fetch('https://fakestoreapi.com/products')
            .then(res=>res.json())
            .then(json=> setProducts(json))
    }, []);

    return (
        <>
            <Navbar />
            <h1>Hi! This is the shop page.</h1>
            { products &&
                products.map(product => <Product data={product} key={product.id} />)
            }
        </>
    )
}