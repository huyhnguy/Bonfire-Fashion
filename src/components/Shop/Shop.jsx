import Navbar from "../Navbar"
import Product from "../Product/Product"
import { useEffect, useState } from "react"
import styles from "./Shop.module.css"

export default function Shop() {
    const [products, setProducts] = useState(null);
    const [productNumber, setProductNumber] = useState(localStorage.length);

    useEffect(() => {
        fetch('https://fakestoreapi.com/products')
            .then(res=>res.json())
            .then(json=> setProducts(json))
    }, []);

    function handleClick(id, quantity) {
        for (let i = 0; i < localStorage.length; i++) {
            const currentId = Number(localStorage[i].substring(0, localStorage[i].indexOf('-')));
            if (id === currentId) {
                const currentQuantity = Number(localStorage[i].substring(localStorage[i].indexOf('-') + 1));
                const newQuantity = currentQuantity + quantity;

                return localStorage.setItem([i], `${id}-${newQuantity}`);
            }
        }

        localStorage.setItem(productNumber, `${id}-${quantity}`);
        setProductNumber(localStorage.length);
    }

    return (
        <>
            <Navbar/>
            <h1 className={styles.title}>Hi! This is the shop page.</h1>
            <div className={styles.shop}>
                { products &&
                    products.map(product => <Product data={product} key={product.id} handleClick={handleClick}/>)
                }
            </div>
        </>
    )
}