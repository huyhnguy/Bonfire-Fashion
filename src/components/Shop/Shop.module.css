.shop {
    display: flex;
    flex-wrap: wrap;
    gap: 1rem;
    justify-content: space-evenly;
}

.title {
    text-align:center;
}