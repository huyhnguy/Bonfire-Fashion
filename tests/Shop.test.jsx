import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Shop from "../src/components/Shop/Shop";
import { BrowserRouter } from "react-router-dom";

describe("Shop component", () => {
    it("Shop renders", () => {
        const { container } = render(
        <BrowserRouter>
            <Shop />
        </BrowserRouter>);
        expect(container).toMatchSnapshot();
    });

});