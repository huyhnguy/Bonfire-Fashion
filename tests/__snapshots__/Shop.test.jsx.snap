// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Shop component > renders all products 1`] = `
<div>
  <div
    role="navigation"
  >
    <div>
      <a
        href="/"
      >
        Home
      </a>
    </div>
    <div>
      <a
        href="/shop"
      >
        Shop
      </a>
    </div>
    <div>
      <a
        href="/cart"
      >
        Shopping Cart
         (
        0
        )
      </a>
    </div>
  </div>
  <h1
    class="_title_008a62"
  >
    Hi! This is the shop page.
  </h1>
  <div
    class="_shop_008a62"
  />
</div>
`;
